apps/moodytunes/tasks.py

```diff
@@ -130,7 +130,27 @@ def add_songs_to_playlist(self, auth_code, playlist_id, songs, spotify):
             spotify.add_songs_to_playlist(auth_code, playlist_id, batch)
 
     @update_logging_data
-    def run(self, auth_id, playlist_name, songs, *args, **kwargs):
+    def upload_cover_image(self, auth_code, playlist_id, cover_image_filename, spotify, **kwargs):
+        """
+        Upload custom cover image for playlist. If any errors were encountered it will just fail
+        silently.
+
+        :param auth_code: (str) SpotifyUserAuth access_token for the given user
+        :param playlist_id: (str) Spotify ID of the playlist to be created
+        :param cover_image_filename: (str) Filename of cover image as a file on disk
+        :param spotify: (spotify_client.SpotifyClient) Spotify Client instance
+        """
+        try:
+            spotify.upload_image_to_playlist(auth_code, playlist_id, cover_image_filename)
```

Comment:
Should we try to delete the file once we've uploaded the image to Spotify? Otherwise we'll end up with a lot of image files on disk that could cause problems if the disk fills up.

Reply:
There's an issue in using the `tmp/` directory to write these files in terms of file permissions. When the files are written, they belong to the user who is writing the files (either `vagrant` if the test server is running or `gunicorn` if the app server is being used.) The files will have their group set to the owner of the file, preventing another user from deleting the file. For example, if `gunicorn` ends up writing the file then the `celery` user will not have proper permissions to delete it. 

If we move the directory to which the files are written to `/var/mtdj/` and set read/write permissions properly on the written files this should alleviate the issue. We can set permissions on the directory to persist the group (sticky bit) for files written to this directory and ensure that all users in the `worker` group can read and write these files. This should ensure the user running the application as well as the `celery` user can properly read the file and write to it, for writing the file contents in the application and deleting the file from the task user.